feat: FIT-615: FSM Entity History endpoint

mcanu opened the pull request:
<!-- Follow Merge -->
This PR was automatically generated via Follow Merge. [Workflow Run](https://github.com/HumanSignal/label-studio/actions/runs/19343724668)
Please ensure that all linked upstream Pull Requests are merged before proceeding with this one.
- https://github.com/HumanSignal/label-studio-sdk/pull/672
<!-- Follow Merge -->

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

mcanu committed:
Created history endpoint. Removed limit from state_history method

mcanu committed:
Added url correctly

mcanu committed:
Added new permission

mcanu committed:
Added tests

mcanu committed:
Added ordering by id

bmartel commented on label_studio/pytest.ini:
```diff
@@ -1,6 +1,7 @@
 [pytest]
 DJANGO_SETTINGS_MODULE = core.settings.label_studio
 python_files = tests.py test_*.py *_tests.py
+addopts = -p no:warnings
```
What does this do?

mcanu replied:
removes the huge list of warnings we have when running pytest by default. I already added that a while back to LSE but now for LSO. We usually check warnings when updating libraries, but 99% of the time we ignore them and they were hiding test errors in the terminal.

bmartel commented:
@mcanu there are the same failing checks in each test suite https://github.com/HumanSignal/label-studio/actions/runs/19309538650/job/55225870659?pr=8803#step:9:342

Not really sure what the point of those tests are, but apparently we test for it.

mcanu commented:
> @mcanu there are the same failing checks in each test suite https://github.com/HumanSignal/label-studio/actions/runs/19309538650/job/55225870659?pr=8803#step:9:342
> 
> Not really sure what the point of those tests are, but apparently we test for it.

Its the API docs I'm missing, working on it

mcanu committed:
Added API docs

mcanu committed:
Made endpoint internal

mcanu committed:
Added factories

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19343509540

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-fit-615'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19343509540

mcanu committed:
Updated all_urls.json

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19343724668

mcanu committed:
Updated all_urls.json

mcanu committed:
Fix test

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio-enterprise/actions/runs/19346732821
